### ISSUE TITLE
 Replacing the hardware restet with the #CZ command, possible?

### ISSUE THREAD

**Issue by rigelinorion** [tomaskovacik/OVC3860#15]
Hello. I was wondering if it is possible to replace the hardware reset with the AT # CZ command. It would simplify things a bit. The second question. In the diagrams it is shown that the reset is connected to the 1.8V voltage output through the 10k resistor. Can pulling up the power supply damage the module or cause malfunction?

**Comment by tomaskovacik** [comment]
question is, if improperly  restarted module will understant serial data 

which schematics?

pull-up over voltage divider to say 5V is fine

**Comment by rigelinorion** [comment]
Thanks for the answer. After checking the application. Entering in monfigMode after sending the #CZ command not working. Another question ... You use different commands in frame for reading, for example ... in OVC3860 :: readName you send a frame uint8_t Date [4] = {0x11, 0xc7, 0x00, 0x10}; and in OVC3860 :: readBTAddress you send uint8_t Date [4] = {0x10, 0x18, 0x00, 0x6} ;. Does this mean that the 0x11 command is used to receive a string of characters and the 0x10 command of a number?

**Comment by tomaskovacik** [comment]
https://github.com/tomaskovacik/OVC3860/wiki/PSKey-configuration-mode-for-OVC3860-based-bluetooth-audio-modules

**Comment by rigelinorion** [comment]
Now I understand. The address is the address in the eeprom memory or the option index as it is in the OVC3860 RevE program?

**Comment by rigelinorion** [comment]
I just checked. It seems  that the address is in memory. So now I will ask if there is a list of addresses of individual options or should you calculate it yourself?

**Comment by tomaskovacik** [comment]
read this https://github.com/tomaskovacik/kicad-library/blob/master/library/datasheet/OVC3860_based_modules/OVC3860_RevE_PSKeys_Setting_v1.2.pdf

I updated wiki page with link to this document.

everything what is in this library is based on my investigation in dokuments which I placed here: https://github.com/tomaskovacik/kicad-library/tree/master/library/datasheet/OVC3860_based_modules

**Comment by rigelinorion** [comment]
Thank you very much for the answers. I found the addresses of the most-used options on foot :). Maybe if I find the time, I will make a map of addresses of all options. Thank you once again for the wonderful work you have done and for sharing it on the github. Greetings.

**Comment by tomaskovacik** [comment]
thx!

that will be nice ;)

**Comment by tomaskovacik** [comment]
i'm closing this, if you start doing list/map of all options drop me line here, please

**Comment by rigelinorion** [comment]
Take a look...
https://github.com/rigelinorion/OVC3860-MemoryMap/blob/master/OVC3860_memmap.txt

**Comment by tomaskovacik** [comment]
nice nice :) 

so next step:
- create header file with #defined , so it can be included and used in other code?
- create bunch of useful functions (just for useful options)?, in separate file, so it will be included and easy removed if someone wanna have minimalist code

add suport for 
- reading
- writing
- restoring to default

for these keys? 

will it be usefull?

**Comment by rigelinorion** [comment]
Yes. Now I'm working on the cdc blaupunkt emulator and I just want to use the OVC3860 module as a wireless phone set and listening to music. I just missed the possibility of configuring the bluetooth module using the radio panel. Thanks to you, I moved forward. Once I have put together everything, I will share the header files.

**Comment by tomaskovacik** [comment]
Do not forget to use diferrencial preamp, or at least connect audio ground via capacitor, it has 0.9V offset to power ground

**Comment by rigelinorion** [comment]
Tomas. I was thinking about using the #MX and #MW commands to read and write data in the eeprom. I tried to read with the help of my memory map but it reads different values ​​from the real one. Entering the address in the form # MX07 does not read the contents of the cell with this address. In the producer's note, the address format is given in this way # MX08001AC4. I wonder how to give an address for example 0x07 using this format. Do you have any idea? 
PS. I used a DC / DC converter with full isolation and eliminated the need for a differential amplifier. He works without the slightest noise. I had a bit of a problem with the microphone, very big distortions were. It turned out that the use of a microphone with a bandwidth exceeding 100Hz solved the problem completely.

**Comment by tomaskovacik** [comment]
regarding memory read, i thing that eeprom address  start is not at 0x00, you need to find offset where eeprom start, if it use same memory space

please provide parts numbers 

edit:

in pskey config tool is this:

473		0x2C8	code_src							0x0392	

tray it as offset of eeprom 

or:   0x2CA	code_des							0x08006900
